GCC 14 Build Failure: speexdsp/fftwrap.c: passing argument 2 of ‘spx_fft’ from incompatible pointer type

### Describe the bug

DOSBox-X 2024.03.01 fails to build with GCC 14 using the `build-sdl2` script.  This bug has been reproduced on both Gentoo (https://bugs.gentoo.org/921909) and Fedora 40 (pre-release now, but this issue should be caused by GCC 14 rather than a pre-release distro version).

```
speexdsp/fftwrap.c: In function ‘spx_fft_float’:
speexdsp/fftwrap.c:388:19: error: passing argument 2 of ‘spx_fft’ from incompatible pointer type [-Wincompatible-pointer-types]
  388 |    spx_fft(table, _in, _out);
      |                   ^~~
      |                   |
      |                   spx_word16_t * {aka short int *}
speexdsp/fftwrap.c:102:34: note: expected ‘float *’ but argument is of type ‘spx_word16_t *’ {aka ‘short int *’}
  102 | void spx_fft(void *table, float *in, float *out)
      |                           ~~~~~~~^~
speexdsp/fftwrap.c:388:24: error: passing argument 3 of ‘spx_fft’ from incompatible pointer type [-Wincompatible-pointer-types]
  388 |    spx_fft(table, _in, _out);
      |                        ^~~~
      |                        |
      |                        spx_word16_t * {aka short int *}
speexdsp/fftwrap.c:102:45: note: expected ‘float *’ but argument is of type ‘spx_word16_t *’ {aka ‘short int *’}
  102 | void spx_fft(void *table, float *in, float *out)
      |                                      ~~~~~~~^~~
speexdsp/fftwrap.c: In function ‘spx_ifft_float’:
speexdsp/fftwrap.c:424:20: error: passing argument 2 of ‘spx_ifft’ from incompatible pointer type [-Wincompatible-pointer-types]
  424 |    spx_ifft(table, _in, _out);
      |                    ^~~
      |                    |
      |                    spx_word16_t * {aka short int *}
speexdsp/fftwrap.c:120:35: note: expected ‘float *’ but argument is of type ‘spx_word16_t *’ {aka ‘short int *’}
  120 | void spx_ifft(void *table, float *in, float *out)
      |                            ~~~~~~~^~
speexdsp/fftwrap.c:424:25: error: passing argument 3 of ‘spx_ifft’ from incompatible pointer type [-Wincompatible-pointer-types]
  424 |    spx_ifft(table, _in, _out);
      |                         ^~~~
      |                         |
      |                         spx_word16_t * {aka short int *}
speexdsp/fftwrap.c:120:46: note: expected ‘float *’ but argument is of type ‘spx_word16_t *’ {aka ‘short int *’}
  120 | void spx_ifft(void *table, float *in, float *out)
      |                                       ~~~~~~~^~~
```

### Steps to reproduce the behaviour

This issue can be reproduced in a Fedora 40 Docker container.  I used Podman, but Docker should work the same.

```console
$ podman run -it fedora:40 /bin/bash
```

Inside the container, run:
```console
# dnf group install "C Development Tools and Libraries"
# dnf install SDL_net-devel SDL2_net-devel libxkbfile-devel ncurses-devel libpcap-devel libslirp-devel libpng-devel fluidsynth-devel freetype-devel nasm
# dnf install libXext-devel
# dnf install git
# git clone --depth 1 --branch dosbox-x-v2024.03.01 https://github.com/joncampbell123/dosbox-x.git
# cd dosbox-x/
# ./build-sdl2
```

### Expected behavior

The build completes without any error.

### What operating system(s) this bug have occurred on?

fedora:40 Docker image (https://hub.docker.com/layers/library/fedora/40/images/sha256-5f48188999a2ef1bebb855874e5eb5913efe8a5e7afc77006b5e42bc2cbe9bc2)

### What version(s) of DOSBox-X have this bug?

2024.03.01 SDL2, tag dosbox-x-v2024.03.01

### Used configuration

```ini
$ gcc --version
gcc (GCC) 14.0.1 20240228 (Red Hat 14.0.1-0)
Copyright (C) 2024 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$ cat /etc/os-release 
NAME="Fedora Linux"
VERSION="40 (Container Image Prerelease)"
ID=fedora
VERSION_ID=40
VERSION_CODENAME=""
PLATFORM_ID="platform:f40"
PRETTY_NAME="Fedora Linux 40 (Container Image Prerelease)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:40"
DEFAULT_HOSTNAME="fedora"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f40/system-administrators-guide/"
SUPPORT_URL="https://ask.fedoraproject.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=40
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=40
SUPPORT_END=2025-05-13
VARIANT="Container Image"
VARIANT_ID=container
```


### Output log

```text
Please see "full build log" in "Additional information"
```


### Additional information

[Full build log](https://github.com/joncampbell123/dosbox-x/files/14469922/build.log)


### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

--- a/CHANGELOG
+++ b/CHANGELOG
@@ -4,10 +4,11 @@ Next:
     0xD000 by default, but apparently there are games that require the EMS
     page frame to exist at 0xC000. For these games, you can now set under the
     [dos] section "ems frame=C000". (joncampbell123)
-  - Restore libslirp support for 32-bit MinGW CI builds which was temporarily
+  - Restored libslirp support for 32-bit MinGW CI builds which was temporarily
     dropped in 2024.03.01 release. Also since MinGW plans to gradually phase
     out 32-bit support, added code to manually build on our own. (maron2000)
-  - Fix build errors of Windows installers (maron2000) 
+  - Fixed build errors of Windows installers (maron2000) 
+  - Fixed compile error of speexdsp/fftwrap.c on gcc-14 (maron2000)
 
 2024.03.01
   - If an empty CD-ROM drive is attached to IDE emulation, return "Medium Not

--- a/configure.ac
+++ b/configure.ac
@@ -379,6 +379,7 @@ AC_CHECK_CXXFLAGS([ -Wextra ])
 AC_CHECK_CXXFLAGS([ -Wunused ])
 AC_CHECK_CXXFLAGS([ -pedantic ])
 AC_CHECK_CXXFLAGS([ -Wno-error=format-security ])     # imfc.cpp and later versions of GCC, see https://github.com/joncampbell123/dosbox-x/issues/4436
+AC_CHECK_CXXFLAGS([ -Wno-error=incompatible-pointer-types ]) # required to compile speexdsp/fftwrap.c with GCC 14
 #AC_CHECK_CXXFLAGS([ -Wconversion ])      DO NOT ENABLE. THIS WARNING IS WAY TOO PEDANTIC TO BE USEFUL, EXCEPT FOR SPECIFIC CASES
 #AC_CHECK_CXXFLAGS([ -Wsign-conversion ])
 AC_CHECK_CXXFLAGS([ -Wlogical-op ])

